lib/migrator/index.js

```diff
@@ -113,40 +114,38 @@ const prepareMigration = async (kitDependency, projectDirectory) => {
     })
 }
 
+// Special case app/views/layout.html, as it has moved in prototype
+// starter files, but we don't want to move for existing users
+const upgradeLayoutIfUnchanged = async () => {
+  await upgradeIfUnchanged(
+    ['app/views/layout.html'],
+    'app/views/layouts/main.html',
+    () => deleteIfUnchanged([
+      'app/views/includes/head.html',
+      'app/views/includes/scripts.html'
+    ]))
+}
+
 const migrate = async () => {
   await logger.setup()
 
   try {
     await Promise.all([
+      migrateConfig('app/config.js'),
       prepareAppRoutes('app/routes.js'),
       prepareSass('app/assets/sass/application.scss'),
       deleteUnusedFiles(filesToDelete),
       deleteUnusedDirectories(directoriesToDelete),
-      upgradeIfUnchanged([
-        ...filesToUpdateIfUnchanged.map(filePath => ({ filePath, action: 'copyFromKitStarter' })),
-        ...filesToDeleteIfUnchanged.map(filePath => ({ filePath, action: 'delete' }))
-      ]),
-      upgradeIfUnchanged([
-        // Special case app/views/layout.html, as it has moved in prototype
-        // starter files, but we don't want to move for existing users
-        { filePath: 'app/views/layout.html', starterFilePath: 'app/views/layouts/main.html', action: 'copyFromKitStarter' }
-      ])
+      upgradeIfUnchanged(filesToUpdateIfUnchanged),
+      upgradeLayoutIfUnchanged(),
+      deleteIfUnchanged(filesToDeleteIfUnchanged),
+      deleteIfUnchanged(patternsToDeleteIfUnchanged)
     ])
 
-    await Promise.all(directoriesToDeleteIfEmpty.map(async (dirPath) => {
-      const reporter = await addReporter(`Remove empty directory ${dirPath}`)
-      // Only report if a directory is empty
-      try {
-        if (await deleteDirectoryIfEmpty(dirPath)) {
-          await reporter(true)
-        }
-      } catch (e) {
-        await reporter(false)
-      }
-    }))
-
-    // migrate config last so that it will indicate the migration is complete
-    await migrateConfig('app/config.js')
+    await Promise.all([
+      await removeOldPatternIncludesFromSassFile(patternsToDeleteIfUnchanged, 'app/assets/sass/application.scss'),
+      await deleteEmptyDirectories(directoriesToDeleteIfEmpty)
+    ])
```

Comment:
Nice, good to see the logic getting a bit cleaner 🙌

Reply:
Thanks 👍